allocator_test.go

```diff
@@ -179,12 +181,77 @@ func TestGIVENCapacityConstraintWHENInsufficientNodeCapacityTHENAllocationFails(
 	require.Nil(t, allocation)
 }
 
-func buildCluster(clusterSize int64, nodeCapacities []int64) []allocator.Node {
+func TestGIVENTaggingConstraintWHENTagsExistOnNodesTHENAllocationSucceeds(t *testing.T) {
```

Comment:
This test is a bit unreadable. I'd use simpler tags ("a", "b", "c") and fewer nodes/ranges.